### PR TITLE
New: Category search component

### PR DESCRIPTION
**Pull request by patrick-ivers**
#### Changes
- Add category search component for searching marketplace products by category

#### Trello
https://trello.com/c/MJpvDNof/5503-marketplace-category-search-category-filter-country-filter-3d

#### Notes
I've disabled linting for the entire `CategorySearch.scss` file, as the sass-lint module doesn't allow disabling linting for lines. Style linting needs to be disabled for this component as it requires highly specific overrides of the react-select plugin that cannot be achieved if we adhere to the "max 3 levels nesting" and "selectors must be nested" rules.

#### Screenshots
category-search

### CODE REVIEW

File: src/components/Main.js [hunk 1]
```diff
@@ -162,6 +203,10 @@ class AppComponent extends React.Component {
     this.setState({ startDate: newValue });
   }
 
+  performCategorySearch() {
+    console.log(`Searching "${this.state.categorySearchString}"...`);
```
Comment (OndrejCholeva) [inline src/components/Main.js]: console.log?

File: src/components/adslotUi/CategorySearchComponent.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  categoryOptions,
+  onCategorySelect,
+  searchString,
+  searchPlaceholder,
+  onSearchStringChange,
+  onSearchButtonClick,
+}) => {
+  const placeholder = searchPlaceholder || 'Search';
+  const onCategorySelectBound = (option) => onCategorySelect(option.value);
+  const onSearchStringChangeBound = (evt) => onSearchStringChange(evt.target.value);
```
Comment (OndrejCholeva) [inline src/components/adslotUi/CategorySearchComponent.js]: evt is event?

File: src/components/Main.js (hunk 1 above)
Comment (patrick-ivers) [inline src/components/Main.js]: This method doesn't need to do anything. It just needs to be called. I could keep it empty or throw a _.noop() in there if that makes any difference.

Comment (OndrejCholeva) [inline src/components/Main.js]: Ah, it is an example... I guess console.log is fine then.

File: karma.conf.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ module.exports = function configureKarma(config) {
     port: 8080,
     captureTimeout: 60000,
     frameworks: ['phantomjs-shim', 'mocha', 'chai'],
+    // logLevel: config.LOG_DEBUG,
```
Comment (vinteo) [inline karma.conf.js]: do we need this?

**Comment by vinteo** [comment]
LGTM

File: karma.conf.js (hunk 3 above)
Comment (patrick-ivers) [inline karma.conf.js]: I left it in there so we an easily uncomment it if we need to debug tests. I can take it out if you like.

**Comment by pphminions** [comment]
LGTM 🔍

**Comment by patrick-ivers** [comment]
@mcslash

**Comment by mcslash** [comment]
After those changes LGTM. Looking schmick my friend. 💃

**Comment by omgaz** [comment]
Ignoring codecov failure. All good. Merging 🍜

**Pull request merged**